go-controller/pkg/ovn/gateway_init.go

```diff
@@ -314,7 +314,14 @@ func gatewayInit(nodeName string, clusterIPSubnet []*net.IPNet, hostSubnets []*n
 				return fmt.Errorf("failed to create default SNAT rules for gateway router %s: %v",
 					gatewayRouter, err)
 			}
-
+			// delete the existing lr-nat rule first otherwise gateway init fails
+			// if the external ip has changed, but the logical ip has stayed the same
+			stdout, stderr, err = util.RunOVNNbctl("--if-exists", "lr-nat-del",
+				gatewayRouter, "snat", entry.String())
+			if err != nil {
+				return fmt.Errorf("failed to delete SNAT rule for gateway router %s, "+
+					"stdout: %q, stderr: %q, error: %v", gatewayRouter, stdout, stderr, err)
+			}
 			stdout, stderr, err = util.RunOVNNbctl("--may-exist", "lr-nat-add",
 				gatewayRouter, "snat", externalIP.String(), entry.String())
 			if err != nil {
```

Comment:
@abhat can we please mix both the invocations above into one call so that they are in one transaction?

```
stdout, stderr, err = util.RunOVNNbctl(
                               "--if-exists", "lr-nat-del", gatewayRouter, "snat", entry.String()
                              "--",  "lr-nat-add", gatewayRouter, "snat", externalIP.String(), entry.String())
```